File: Data Structures & Algorithms/is-palindrome/submission-3.java
class Solution {
    public boolean isPalindrome(String s) {
        String rev="",rev1="";
        String s1=s.toLowerCase();
        for(char c:s1.toCharArray())
        {
            if(Character.isLetter(c)||(Character.isDigit(c)))
            {
                rev=rev+c;
                rev1=c+rev1;
            }
        }
        if(rev.equals(rev1))
        {
            return true;
        }
        else
        {
            return false;
        }
    }
}

File: Data Structures & Algorithms/climbing-stairs/submission-3.py
class Solution:
    def climbStairs(self, n: int) -> int:
        s={}
        def dp(n,s):
            if n==0:
                return 1
            if n==1:
                return 1
            if n in s:
                return s[n]
            s[n]=dp(n-1,s)+dp(n-2,s)
            return s[n]
        return dp(n,s)